### PR TITLE
docs: add a tutorial for evolinstruct

### PR DESCRIPTION
**Pull request by sdiazlor**
Tutorial for Evol-instruct

NOTES: 

- After getting the feedback, I will run it with more samples (at the moment only 3 were used for testing purposes) and push the datasets to HF.
- I'll also rethink a bit the title

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
@sdiazlor 
I would reason from a feedbackdataset/customdataset and copy back and forth between that one and a datasets.Dataset after filtering in each iteration. 

Perhaps an image of the process might be nice to add.

The title is okay, perhaps we can reference it as a paper implementation or something.

Not sure about the pricing for 50K instruction improvements x4 (evolutions) + ratings API calls but perhaps we might run an estimation expense computation before. Also because the dataset ideally should also be used to fine-tune a model to show improved performance.

**Review by dvsrepo — CHANGES_REQUESTED** [review]
The import argilla at the beginning fails I would remove it from there.

Also the ARGILLA env variables gives the impression that you need argilla for running the whole tutorial, similar for the HF_TOKEN. I'd recommend to keep the start of tutorials very lightweight with the very minimum you need (except maybe for imports because it's cleaner).

**Comment by davidberenstein1957** [comment]
@dvsrepo I agree but when initializing the notebook we would need to include re-loading the import because internally, we have a variable `IF_ARGILLA_AVAILABLE` within `distilabel`, which is already set during the original `distilabel` import, therefore we would need to do something like this @sdiazlor https://stackoverflow.com/questions/50339549/google-colab-reload-imported-modules

**Comment by davidberenstein1957** [comment]
Can you include that in tutorials like this one too? https://distilabel.argilla.io/latest/tutorials/clean-preference-dataset-judgelm-gpt/#human-feedback-with-argilla

File: docs/tutorials/create-evol-instruct-dataset.ipynb [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+    "        start = time.time()\n",
+    "        elimination_dataset=elimination_pipe.generate(prepared_dataset, batch_size=8)\n",
+    "        end = time.time()\n",
+    "        print(\"Elapsed\", end - start)\n",
+    "        \n",
+    "        # Save the successful instructions in the pool and prepare the inputs for the next iteration\n",
+    "        new_instructions = []\n",
+    "        \n",
+    "        for row_evolved, row_elimination in zip(evolved_dataset, elimination_dataset):\n",
+    "            if row_evolved['instructions'][0] is None or row_elimination['generations'][0] != \"equal\":\n",
```
[the remark this answers is not in the document]
Comment (sdiazlor) [reply docs/tutorials/create-evol-instruct-dataset.ipynb:384]: Yes and it's an and. But I hadn't pushed the correct changes

**Comment by dvsrepo** [comment]
> @dvsrepo I agree but when initializing the notebook we would need to include re-loading the import because internally, we have a variable `IF_ARGILLA_AVAILABLE` within `distilabel`, which is already set during the original `distilabel` import, therefore we would need to do something like this lateron in te notebook @sdiazlor https://stackoverflow.com/questions/50339549/google-colab-reload-imported-modules

You are absolutely right, forget my comment because I experienced that first hand and it's kind of terrible when it happens, it's fine like that

**Comment by davidberenstein1957** [comment]
Hi @sdiazlor, there is a slight nuance that is missing in the current implementations. And evolution should run the evol complexity and quality sequentially n times. Currently, we run the complexity n times and quality n times after that.
it should be
complexity->quality->complexity->etc.

**Comment by sdiazlor** [comment]
@davidberenstein1957 I updated the tutorial and there I also pointed out my logic, as if we follow the Deita idea regarding Evol-Quality, they can not be completely sequential.

**Review by davidberenstein1957 — COMMENTED** [review]
Great. In the 1.0.0 we will split them up, but for now, it is good and can be merged.

**Pull request merged**